TypeCobol.LanguageServer/Completion Factory/CompletionFactory.cs

```diff
@@ -424,15 +424,19 @@ public static IEnumerable<CompletionItem> GetCompletionForQualifiedName(Position
                     foreach (var variable in possibleVariables)
                     {
                         var children = new List<Node>();
-                        if (variable.Children != null && variable.Children.Count > 0) //It's a variable with levels inside
-                            children.AddRange(variable.Children);
-                        else //It's a typed variable, we have to search for children in the type
+
+                        //if it's a typed variable, propose 1st children of the type
+                        if (variable.TypeDefinition != null)
                         {
-                            var typeChildren = GetTypeChildren(node.SymbolTable, variable);
-                            if (typeChildren != null)
-                                children.AddRange(typeChildren.Where(t => t.Name != null || t.Children.Where(u => u.Name != null) != null));
+                            children.AddRange(variable.TypeDefinition.Children.Where(t => t.Name != null || t.Children.Where(u => u.Name != null) != null));
```

Comment:
`t.Name != null || t.Children.Where(u => u.Name != null) != null` is always `true` because LINQ Where method always produces a non-null result. So in its current form, this line is equivalent to `children.AddRange(variable.TypeDefinition.Children);`.
Is this intended ? I know it was written originally but this is different now because you're not using the `GetTypeChildren` method so :
- before your modification : every node returned by GetTypeChildren were selected
- after your modification : every child from variable is selected

Reply:
So after reviewing the code of `GetTypeChildren`, I think it's ok. It is equivalent to selecting all children from the type, you can change the line to `children.AddRange(variable.TypeDefinition.Children);`

Reply:
done
https://github.com/TypeCobolTeam/TypeCobol/pull/1605/commits/99ff165773be105e3be39d1adc6750d3d026e678